kotlinx.fuzz.gradle/src/test/kotlin/kotlinx/fuzz/gradle/test/FuzzConfigBuilderTest.kt

```diff
@@ -0,0 +1,59 @@
+package kotlinx.fuzz.gradle.test
+
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.fuzz.gradle.KFuzzConfigBuilder
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+
+object FuzzConfigBuilderTest {
+    @Test
+    fun `not initialize something`() {
+        @Suppress("EMPTY_BLOCK_STRUCTURE_ERROR")
+        assertThrows<Throwable> { KFuzzConfigBuilder.build {} }
+    }
+
+    @Test
+    fun `0 maxSingleTargetFuzzTime fails`() {
+        assertThrows<IllegalArgumentException> {
+            KFuzzConfigBuilder.build {
+                maxSingleTargetFuzzTime = 0.seconds
+                instrument = emptyList()
+            }
+        }
+    }
+
+    @Test
+    fun `enough set`() {
+        assertDoesNotThrow {
+            KFuzzConfigBuilder.build {
+                instrument = listOf("1", "2")
+                maxSingleTargetFuzzTime = 30.seconds
+            }
+        }
+    }
+
+    @Test
+    fun `all set`() {
+        assertDoesNotThrow {
+            KFuzzConfigBuilder.build {
+                fuzzEngine = "engine"
+                hooks = true
+                keepGoing = 339
+                instrument = listOf()
+                customHookExcludes = listOf("exclude")
+                maxSingleTargetFuzzTime = 1000.seconds
+            }
+        }
+    }
+
+    @Test
+    fun `can't set option twice`() {
+        assertThrows<IllegalStateException> {
+            KFuzzConfigBuilder.build {
+                fuzzEngine = "once"
+                fuzzEngine = "twice"
+            }
+        }
+    }
```

Comment:
do we also need to test whether the config is properly stored and loaded from system properties? (because right now I'm not even sure how to do it...)